src/transaction/log_manager.c

```diff
@@ -3730,6 +3734,24 @@ log_append_assigned_mvccid (THREAD_ENTRY * thread_p, MVCCID mvccid)
   (void) prior_lsa_next_record (thread_p, node, tdes);
 }
 
+void
+log_append_schema_modification_lock (THREAD_ENTRY * thread_p, const OID * classoid)
+{
+  assert (!OID_ISNULL (classoid));
+
+  LOG_TDES *tdes = LOG_FIND_CURRENT_TDES (thread_p);
+  assert (tdes != nullptr);
```

Comment:
In old `log_append_..` functions, these cases are treated with a return.
In case this happens in release builds, the server would just crash below in the `prior_lsa_next_record` where `tdes` is expected to have a value.
I think the guideline is that "the server should never crash; it can abort some transactions, but it should never crash".

Share your opinion please.

Reply:
I think that the `log_append_...` functions failing should return some error code that should only affect the requesting transaction (or, if it's a system transaction, should be dealt-with with some retry mechanism or else ...).
But it is out of scope for us.
I personally would return as the older functions do.
Assert has no value in production and would just crash the server.
Up to you. :)

Reply:
Alright, I understand. I will make the modification to return in case the pointer value is null, just like in `log_append_redo_crumbs()`.